### DISCUSSION TITLE
Ability to add conditional attributes?

### DISCUSSION THREAD

**Discussion by frankie2784** [streetcomplete/StreetComplete#4628] [Ideas]
I've come across a number of situations where I'm asked to record, for example 1) a speed limit in a variable speed zone, or 2) street parking or cycle lanes, which are designated at only certain times of day. It would be great to be able to add the ability to tag attributes such as this with an optional time component.

I envision this could be accomplished by adding another option under "Other answers..." such as "Varies by time or day" and using a similar template to how we specify opening hours for businesses.

This could then flow through to OSM via a :conditional tag on the feature in question.

Not sure how complex this would be to implement in practice, and which challenges would make sense to include this feature on. But the examples I mentioned above would be a good starting point to test such a feature.

**Comment by mnalis** [comment]
I personally would just **leave a note with a picture**, to be solved in desktop editor. Because:

- they would need **very complex UI**, and StreetComplete is targeted for **simplicity** and quick solving of quests (also, it would need a lot of work to implement). Such complexity and feature-fullness might be a better match for [Vespucci](https://wiki.openstreetmap.org/wiki/Vespucci)
- those quests themselves are _already_ complex for average SC user (some of them to the point that they are disabled by default)
- note that those special conditional restrictions should also be shown to the user on _resurvey quests_ (or the data will simply become wrong with time), which needs even more UI complexity 
- even if fully implemented as suggested, it would only solve one subset of the problem, and leave majority of it unsolved - note that [Conditional restrictions](https://wiki.openstreetmap.org/wiki/Conditional_restrictions) are not used only for `times of day`, but also by `type of vehicle`, `max weight`, `stay duration`, `access type`, `length of the vehicle` etc. It also support boolean algebra on combination of them! 
- even such full-fledged ultra-complex conditional restrictions are not enough to map the reality in many cases - e.g. even restricted to just simple time of day, quite complex `opening_hours` format (as used in conditional restrictions) is [unable to map simple and common cases](https://help.openstreetmap.org/questions/72235/opening_hours-differs-on-even-odd-days)
- Although I love mapping detail just for the sake of having it mapped, one should also consider the **use case**: how important that data actually is in each case, how widespread the use is, and most importantly what data consumers are **_actually using that data_** ? (yes, one also consider the opposite: unless the data is mapped, it is not likely to be supported by tools; but reducing amount of work for on-the-ground SC surveyor to just taking a picture and having it be added later and easier on desktop editor is probably preferable IMHO)

**Reply by frankie2784** [reply]
Hey, mnalis thanks for your reply. A couple of thoughts.

I'm going to narrow my suggestion down to variable street signs for the moment as thinking through the parking and bike lane examples, the conditionals can get complex quick for parking, and in the case of cycleways, it really is an edge case.

- On the complexity of the UI I'm not sure if it would need to be much more complex than the opening times UI, just with an additional step to ask what the speed limit is during the elected period(s) and the default limit at other times. But I'm no expert, so I'd have to leave that call to the team.

- I don't see the need for it to add complexity for regular users or lead them to turn off the speed quest as the option would be tucked away in the "Other Answers..." section.

- On resurveys, again, not sure if it needs to be much more complex than the opening times UI. Again, a matter for the experts to debate.

- On the broad nature of conditional tags, agreed. However, SC is not trying to solve for every use case but the most common. The conditional tag is most commonly used for maxspeed by a large margin according to taginfo.

- The example you gave about alternating work days is not really representative of speed zone conditionals in virtually all cases that I'm aware of. Speed zones are much more consistent, FWIW most variable speed limits in my area are pretty simple - the same speeds as the same times every day.

- On mapping detail, I agree that detail for the sake of detail is often unwise. However maxspeed is an attribute that many OSM users, and SC itself, seems to care about, so making it easy for users to capture this attribute accurately would be a bonus.

- On use cases, third-party journey planning applications use speed limit info to determine preferred view routing options. I'm personally using speed information (alongside many other variables) to determine cycling level of stress to help with infrastructure planning decisions. In terms of use cases in the broader community, I'm really not sure, but the fact that maxspeed appears in the top 30 keys in taginfo suggests it's a highly coveted attribute.

Happy to keep the conversation going.

**Reply by mnalis** [reply]
- as for only catering to most common conditional restriction cases; the `maxspeed quest` you mention, being disabled by default and quite problematic to solve time-wise (one **not only** needs to walk both sides of the street before answering, but is supposed to walk the whole block first to confirm that there is not `zone30` or similar active which encompasses much more than one street but is marked only at edges of the block) seems much less likely to get even easier stuff implemented (unless someone volunteers to write the code) - see e.g. https://github.com/streetcomplete/StreetComplete/issues/4612#issuecomment-1304823161
- on resurveys (as opposed to initial survey), one may not hide that information away in `other answers` for advanced users, but must display it all at once when showing quest to any users.
- for maxspeed, I hear it is common that `hgv` / `bus` have different max speed limits than `motorcar`. If there were also time-of-day variation, that would be additional condition on top of that I guess (whether the same for both or again different for each one). For parkings, the variations which seem common to me look even more complex to me (time of day + vehicle type / delivery vs. nondelivery + variable maxstay for permit owners vs. not)
- I do not doubt that many data consumers use `maxspeed=*`. What I was questioning, is how many of them actually would parse even simplest [conditional restrictions](https://wiki.openstreetmap.org/wiki/Conditional_restrictions) in the form `maxspeed:motorcar:conditional=80 @ (06:00-20:00); 60 @ (20:00-06:00);` which this suggestion would presumably create (not to mention more complex ones)

**Reply by frankie2784** [reply]
Sounds like more hassle than it's worth. Good to know.

**Comment by matkoniecz** [comment]
Note that some quests already support it (parking fee quest, I think, partially supports it)
